webvh/webvh/did/operations_manager.py

```diff
@@ -0,0 +1,496 @@
+from .exceptions import DidCreationError
+from .registration_state import RegistrationState
+from .utils import get_server_info
+
+LOGGER = logging.getLogger(__name__)
+
+WEBVH_METHOD = "did:webvh:0.4"
+ENDORSEMENT_WAIT_TIMEOUT_SECONDS = 2
+ENDORSEMENT_EVENT = "endorsement_response::"
+
+
+class DidWebvhOperationsManager:
+    """DID Webvh Manager class."""
+
+    def __init__(self, profile: Profile) -> None:
+        """Initialize the DID Webvh Manager."""
+        self.profile = profile
+
+    def _all_are_not_none(*args):
+        return all(v is not None for v in args)
+
+    async def fetch_jsonl(self, url):
+        """Fetch a JSONL file from the given URL."""
+        async with ClientSession() as session:
+            async with session.get(url, ssl=False) as response:
+                # Check if the response is OK
+                response.raise_for_status()
+
+                # Read the response line by line
+                async for line in response.content:
+                    # Decode each line and parse as JSON
+                    decoded_line = line.decode("utf-8").strip()
+                    if decoded_line:  # Ignore empty lines
+                        yield json.loads(decoded_line)
+
+    async def _request_identifier(self, server_url, namespace, identifier) -> tuple:
+        """Contact the trust did web server to request an identifier."""
+        async with ClientSession() as session:
+            try:
+                response = await session.get(
+                    server_url,
+                    params={
+                        "namespace": namespace,
+                        "identifier": identifier,
+                    },
+                    ssl=False,
+                )
+            except ClientConnectionError as err:
+                raise DidCreationError(f"Failed to connect to Webvh server: {err}")
+
+            response_json = await response.json()
+            if response.status == http.HTTPStatus.BAD_REQUEST:
+                raise DidCreationError(response_json.get("detail"))
+
+            document = response_json.get("didDocument", {})
+            did = document.get("id")
+
+            proof_options = response_json.get("proofOptions", {})
+            challenge = proof_options.get("challenge")
+            domain = proof_options.get("domain")
+            expiration = proof_options.get("expires")
+
+            if self._all_are_not_none(did, challenge, domain, expiration):
+                return did, challenge, domain, expiration
+            else:
+                raise DidCreationError(
+                    "Invalid response from Webvh server requesting identifier"
+                )
+
+    async def _get_or_create_author_key(self, did):
+        async with self.profile.session() as session:
+            try:
+                key_info_author = await MultikeyManager(session).create(
+                    alg="ed25519",
+                    kid=f"{did}#author",
```

Comment:
We will likely have to revisit this when we look at key rotation, maybe put a todo comment in here, in the webvh this would likely be called the "update Key" or "authorized Key:
https://identity.foundation/didwebvh/next/#authorized-keys